### PR TITLE
Resolves Codecademy/docs Issue#4239

### LINKED ISSUE #4239: [Entry] Type conversion: Creating new concept entry `type-conversion.md`
### Reason/inspiration (optional)

We would like a new term entry on the `Type conversion` term for [Dart](https://www.codecademy.com/resources/docs/dart). The entry should go under the following path `docs/content/dart/concepts/type-conversion/type-conversion.md`.

The entry should describe the term and how it is used within Dart. Also include: 
- Add a "Syntax" section.
- An "Example" section that explains type conversion in Dart.
- Explain how different types of type conversion are there with example code snippets.

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Vino-Siva**
### Description

New term entry for Type Conversion in Dart

### Issue Solved

Closes #4239 

### Type of Change

- Adding a new entry
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Review by helfese — APPROVED** [review]

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 1]
```diff
@@ -0,0 +1,496 @@
...
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+In Dart, **type conversion** refers to the process of converting a value from one data type to another. Dart supports both Implicit and Explicit type conversion.
+
+## Implicit Type Conversion
+
+Implicit type conversion is done automatically, when operations involving different numeric types are performed.
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:20]:
```suggestion
Implicit type conversion is done automatically when operations involving different numeric types are performed.
```

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 2]
```diff
@@ -0,0 +1,496 @@
...
+  print(checkCondition);
+}
+```
+
+The `Shell` output:
+
+```shell
+42
+42 is a string
+```
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:100]: We don't need any specific function explanation in this case. Please remove it.

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 3]
```diff
@@ -0,0 +1,496 @@
...
+
+The `Shell` output:
+
+```Shell
+1000000.1
+5e-7
+```
+
+For special values, returns `NaN` if the input is `NaN`, returns `Infinity`, if input is `Infinity`, and `-Infinity`, if the input is `-Infinity`.
+
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:129]: We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 4]
```diff
@@ -0,0 +1,496 @@
...
+The `Shell` output:
+
+```Shell
+NaN
+Infinity
+-Infinity
+```
+
+An `int` is converted to a decimal representation with no decimal point.
+
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:157]: We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 5]
```diff
@@ -0,0 +1,496 @@
...
+The `Shell` output:
+
+```Shell
+123000000000000000000
+1.23e+22
+```
+
+---
+
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:181]: We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 6]
```diff
@@ -0,0 +1,496 @@
...
+1.000
+256463.516
+2351.44465
+254645466111.000
+1224.00000000000000000000
+5
+```
+
+> **Note:** If the `fractionDigit` value is not in specified range, console output will display a `RangeError`.
+
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:242]: We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 7]
```diff
@@ -0,0 +1,496 @@
...
+## Explicit Type Conversion
+
+Explicit type conversion is done by the programmer using predefined functions or methods to convert values from one type to another. The following are the methods for Explicit type conversion:
+
+* `toString()`[^1] : Converts a value to a string.
+* `toDouble()`[^2] : Converts a value to a double.
+* `toInt()`[^3] : Converts a value to an integer.
+* `parse()`[^4] : Converts a string to a numeric type.
+* `tryParse()`[^5]: Similar to `parse()`, but returns `null` if the conversion fails.
+
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:73]: Please add one example of an explicit function. Follow the ### Example block for the same.

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 8]
```diff
@@ -0,0 +1,496 @@
...
+```
+
+The `Shell` output:
+
+```Shell
+Uncaught Error: RangeError (fractionDigits): Invalid value: Not in inclusive range 0..20: 21
+```
+
+---
+
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:258]: We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 9]
```diff
@@ -0,0 +1,496 @@
...
+```Shell
+1.000e+0
+2.565e+5
+2.35144464661e+3
+2.546e+11
+1.22400000000000000000e+3
+5e+0
+```
+
+---
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:312]: We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 10]
```diff
@@ -0,0 +1,496 @@
...
+1.00
+3e+5
+2.4e+3
+2.55e+11
+1224.0000000000000000
+5.25000000000
+```
+
+---
+
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:362]: We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 11]
```diff
@@ -0,0 +1,496 @@
...
+```
+
+The `Shell` output:
+
+```Shell
+true
+```
+
+---
+
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:412]: We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 12]
```diff
@@ -0,0 +1,496 @@
...
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:450]: We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 13]
```diff
@@ -0,0 +1,496 @@
...
+The `Shell` output:
+
+```Shell
+42
+null
+```
+
+---
+
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:486]: We don't need any specific function explanation or example in this case. Please remove the above-mentioned lines.

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @Vino-Siva 
Thanks for your contribution at Codecademy!

I've mentioned a few suggestions, please make sure you make those. A quick summary of what is needed from this entry:

What is type conversion.
Implicit type conversion explanation, with one example.
Explicit type conversion, with one example. (you can mention the in built functions that are used in explicit type, but don't explain those in detail, just a one line explanation as you have written is fine). 

Please make sure you make the required changes as soon as possible! Thank you!

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 14]
```diff
@@ -0,0 +1,496 @@
...
+## References
+
+[^1]: [toString: Dart Documentation](https://api.dart.dev/stable/3.2.6/dart-core/num/toString.html)
+[^2]: [toDouble: Dart Documentation](https://api.dart.dev/stable/3.2.6/dart-core/num/toDouble.html)
+[^3]: [toInt: Dart Documentation](https://api.dart.dev/stable/3.2.6/dart-core/num/toInt.html)
+[^4]: [parse: Dart Documentation](https://api.dart.dev/stable/3.2.6/dart-core/num/toString.html)
+[^5]: [tryParse: Dart Documentation](https://api.dart.dev/stable/3.2.6/dart-core/num/toString.html)
+[^6]: [toStringAsFixed: Dart Documentation](https://api.dart.dev/stable/3.2.6/dart-core/num/toString.html)
+[^7]: [toStringAsExponential: Dart Documentation](https://api.dart.dev/stable/3.2.6/dart-core/num/toString.html)
+[^8]: [toStringAsPrecision: Dart Documentation](https://api.dart.dev/stable/3.2.6/dart-core/num/toString.html)
```
[the remark this answers is not in the document]
Comment (Vino-Siva) [reply content/dart/concepts/type-conversion/type-conversion.md]: Done.

[the remark this answers is not in the document]
Comment (Vino-Siva) [reply content/dart/concepts/type-conversion/type-conversion.md]: Done

File: content/dart/concepts/type-conversion/type-conversion.md (hunk 12 above)
Comment (Vino-Siva) [reply content/dart/concepts/type-conversion/type-conversion.md]: Done

File: content/dart/concepts/type-conversion/type-conversion.md (hunk 11 above)
Comment (Vino-Siva) [reply content/dart/concepts/type-conversion/type-conversion.md]: Done.

File: content/dart/concepts/type-conversion/type-conversion.md (hunk 10 above)
Comment (Vino-Siva) [reply content/dart/concepts/type-conversion/type-conversion.md]: Done

File: content/dart/concepts/type-conversion/type-conversion.md (hunk 9 above)
Comment (Vino-Siva) [reply content/dart/concepts/type-conversion/type-conversion.md]: Done

**Review by Vino-Siva — COMMENTED** [review]
Thanks for suggestions, made necessary changes as requested.

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+  double doubleValue = .5;
+
+  // Demonstrating implicit type conversion during arithmetic operation
+  double result = intValue + doubleValue;
+
+  print(result);
+}
+```
+
+The `Shell` output:
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:38]:
```suggestion
This produces the following output:
```

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+
+```shell
+10.5
+```
+
+In the example, the `int` intValue (10) is implicitly converted to a `double` (10.0) to perform the addition with doubleValue. The resulting value is 10.5, as shown in the Shell output.
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:44]:
```suggestion
In the example, the `int` intValue (10) is implicitly converted to a `double` (10.0) to perform the addition with doubleValue. The resulting value is 10.5, as shown in the output.
```

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+  var stringValue = intValue.toString(); // Explicit toString() call
+
+  print(stringValue);
+```
+
+The `Shell` output:
+
+```shell
+42
+```
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:70]: You can add a short explanation for the example code above.

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
@Vino-Siva 
Thanks for making the changes, just a couple more changes are required. I request you to kindly make those as soon as possible. Thanks!

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 18]
```diff
@@ -0,0 +1,72 @@
...
+- `tryParse()`: Similar to `parse()`, but returns `null` if the conversion fails.
+
+### Example
+
+```dart
+void main() {
+  var intValue = 42;
+  var stringValue = intValue.toString(); // Explicit toString() call
+
+  print(stringValue);
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:63]:
```suggestion
  print(stringValue);
  }
```

Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:63]: The code has an error, it seems like you have missed the "}"

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
@Vino-Siva 
Please check the explicit example code, there seems to be an error w.r.t "}".

**Review by mamtawardhani — APPROVED** [review]

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 19]
```diff
@@ -0,0 +1,73 @@
+---
+Title: 'Type Conversion'
+Description: 'The process of changing the data type of a value from one type to another.'
+Subjects:
+  - 'Computer Science'
+  - 'Mobile Development'
+  - 'Code Foundations'
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:7]:
```suggestion
  - 'Code Foundations'
  - 'Computer Science'
  - 'Mobile Development'
```

File: content/dart/concepts/type-conversion/type-conversion.md [hunk 20]
```diff
@@ -0,0 +1,73 @@
...
+Tags:
+  - 'Dart'
+  - 'Dart Operators'
+  - 'Data Types'
+  - 'Arithmetic'
```
Comment (mamtawardhani) [inline content/dart/concepts/type-conversion/type-conversion.md:12]:
```suggestion
  - 'Arithmetic'
  - 'Dart'
  - 'Dart Operators'
  - 'Data Types'
```

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @Vino-Siva Could you please arrange the tags and subjects in alphabetical manner as suggested.

**Review by mamtawardhani — APPROVED** [review]

**Review by Sriparno08 — APPROVED** [review]
Looks good 👍

**Pull request merged**